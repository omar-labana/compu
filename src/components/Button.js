import React from 'react';
import PropTypes from 'prop-types';

const Button = (props) => {
    const { buttonName } = props;
    return (
        <button type="button" name="button">{buttonName}</button>
    );
};



Button.propTypes = {
    buttonName: PropTypes.string,
};

export default Button;