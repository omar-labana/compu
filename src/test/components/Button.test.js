import React from 'react';
import '@testing-library/jest-dom';
import Button from '../../components/Button';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

const stl = () => 'red';
const doNothing = () => ''

describe('Button Component', () => {
    let wrapper;
    wrapper = shallow(<Button buttonName="1" st={stl} clickHandler={doNothing} />);

    it('exists', () => {
        expect(wrapper.find('button')).toBeTruthy();
    });

    it('has correct props', () => {
        const propa = wrapper.props()
        expect(propa.type).toBe("button")
        expect(propa.name).toBe("button")
        expect(propa.className).toBe("red")
    });

    it('has to match snap', () => {
        const button = renderer.create(<Button buttonName="1" st={stl} clickHandler={doNothing} />);
        expect(button).toMatchSnapshot();
    });
})