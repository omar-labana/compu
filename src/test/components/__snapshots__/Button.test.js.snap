// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component has to match snap 1`] = `
<button
  className="red"
  name="button"
  onClick={[Function]}
  type="button"
>
  1
</button>
`;
